Adds docs for notRestoredReasons (#27947)
mdn/content: files/en-us/web/api/performancenavigationtiming/notrestoredreasons/index.md

---
title: "PerformanceNavigationTiming: notRestoredReasons property"
short-title: notRestoredReasons
slug: Web/API/PerformanceNavigationTiming/notRestoredReasons
page-type: web-api-instance-property
status:
  - experimental
browser-compat: api.PerformanceNavigationTiming.notRestoredReasons
---

{{APIRef("Performance API")}}{{SeeCompatTable}}

> The **`notRestoredReasons`** read-only property of the {{domxref("PerformanceNavigationTiming")}} interface returns a {{domxref("NotRestoredReasons")}} object providing report data on whether frames present in the current document were blocked from using the [back/forward cache (bfcache)](https://web.dev/articles/bfcache) on navigation, and why.

Reviewer: ```suggestion
The **`notRestoredReasons`** read-only property of the {{domxref("PerformanceNavigationTiming")}} interface returns a {{domxref("NotRestoredReasons")}} object providing report data on reasons why the current document was blocked from using the [back/forward cache (bfcache)](https://web.dev/articles/bfcache) on navigation.
```

Author: updated

> **Note:** `notRestoredReasons` may return `null` despite the navigation type being reported as a back/forward navigation. These circumstances include duplicating a back/forward navigation in a new tab and restoring a back/forward navigation tab after a browser restart. In such cases, some browsers copy the navigation type from the original tab, but as these are not actually back/forward navigations, `notRestoredReasons` returns `null`.

## Examples

A [`PerformanceNavigationTiming`](/en-US/docs/Web/API/PerformanceNavigationTiming) instance can be obtained from features such as [`Performance.getEntriesByType()`](/en-US/docs/Web/API/Performance/getEntriesByType) and [`PerformanceObserver`](/en-US/docs/Web/API/PerformanceObserver).

For example, you could invoke the following function to return all `PerformanceNavigationTiming` objects currently present in the performance timeline and log their `notRestoredReasons`:

```js

> function returnNRR() {
>   const navEntries = performance.getEntriesByType("navigation");
>   for (let i = 0; i < navEntries.length; i++) {
>     console.log(`Navigation entry ${i}`);
>     let navEntry = navEntries[i];
>     console.log(navEntry.notRestoredReasons);
>   }
> }

Reviewer: This may future proof this, but at the moment there is only one so could make this a lot shorter:

```suggestion
function returnNRR() {
  console.log(performance.getEntriesByType("navigation")[0].notRestoredReasons);
}
```

Author: I'd rather just leave it future proofed

Reviewer: No bother. But if we ever change it so the short version doesn’t work then lots and lots of things will break so I doubt that’ll ever happen! But also fine to leave as is.

```

> The `PerformanceNavigationTiming.notRestoredReasons` property returns an object with the following structure, which represents the blocked state of a top-level frame with no embedded child `<iframe>`s:

Reviewer: ```suggestion
The `PerformanceNavigationTiming.notRestoredReasons` property returns an object with the following structure, which provides reasons which preventing the usage of the bfcache. In this example the top-level frame had no embedded child `<iframe>`s:
```

Author: updated

- `src`
  - : A string representing the path to the frame's source (for example `<iframe src="b.html">`). If the frame has no `src`, the value will be an empty string.
- `url`
  - : A string representing the URL of the navigated page.

## Examples

A [`PerformanceNavigationTiming`](/en-US/docs/Web/API/PerformanceNavigationTiming) instance can be obtained from features such as [`Performance.getEntriesByType()`](/en-US/docs/Web/API/Performance/getEntriesByType) and [`PerformanceObserver`](/en-US/docs/Web/API/PerformanceObserver).

For example, you could invoke the following function to return all `PerformanceNavigationTiming` objects currently present in the performance timeline and log their `notRestoredReasons`:

```js
function returnNRR() {
  const navEntries = performance.getEntriesByType("navigation");
  for (let i = 0; i < navEntries.length; i++) {
    console.log(`Navigation entry ${i}`);
    let navEntry = navEntries[i];
    console.log(navEntry.notRestoredReasons);
  }
}
```

The `PerformanceNavigationTiming.notRestoredReasons` property returns an object with the following structure, which represents the blocked state of the top-level frame:

> ```js

Reviewer: ```suggestion
```json
```

Author: Again, I think this is a JS object, not JSON
